no error message optional comparison without `?`

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:  0.2.4 3f1e232** 
**OS: ArchLinux** 

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?**
```v
fn foo() ?int{
	return 0
}

fn main(){
	if foo() > foo(){}
}
```

**What did you expect to see?**
error message 
 
**What did you see instead?**
```
> v run main.v
==================
/tmp/v_1000/main.12731590846751467985.tmp.c:11279: error: invalid operand types for binary operation
/tmp/v_1000/main.12731590846751467985.tmp.c:11279: error: invalid aggregate type for register load
...
==================
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -842,6 +842,9 @@ pub fn (mut c Checker) infix_expr(mut node ast.InfixExpr) ast.Type {
 						}
 					}
 				}
+			} else if left_type.has_flag(.optional) && right_type.has_flag(.optional) {
+				c.error('unwrapped optional cannot be compared in an infix expression',
+					left_right_pos)
 			}
 		}
 		.left_shift {

--- a/vlib/v/checker/tests/infix_compare_optional_err.out
+++ b/vlib/v/checker/tests/infix_compare_optional_err.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/infix_compare_optional_err.vv:6:5: error: unwrapped optional cannot be compared in an infix expression
+    4 |
+    5 | fn main(){
+    6 |     if foo() > foo() {}
+      |        ~~~~~~~~~~~~~
+    7 | }

--- a/vlib/v/checker/tests/infix_compare_optional_err.vv
+++ b/vlib/v/checker/tests/infix_compare_optional_err.vv
@@ -1,0 +1,7 @@
+fn foo() ?int{
+	return 0
+}
+
+fn main(){
+	if foo() > foo() {}
+}

